lib/cli/kit/entry_point.rb

```diff
@@ -0,0 +1,150 @@
+require 'cli/kit'
+require 'cli/ui'
+require 'English'
+
+module CLI
+  module Kit
+    class EntryPoint
+      # Interface methods: You may want to implement these:
```

Comment:
The abstraction feels a little rough here but four miscellaneous subclass/interface methods seems tolerable.